landing-page/content/common/how-to-release.md

```diff
@@ -264,33 +264,33 @@ Thanks to everyone for contributing!
 ### Documentation Release
 
 Documentation needs to be updated as a part of Iceberg release after a release candidate is passed.
-The commands described below assume the `iceberg-docs` repository and `iceberg` repository are in the same parent directory locally, 
-and the release manager is executing commands in the `iceberg` repository.
+The commands described below assume the `iceberg-docs` repository and `iceberg` repository are in the same parent directory locally,
+and the release manager is executing commands in the `iceberg` repository. Note that all changes in `iceberg` need to happen against the `master` branch
+and changes in `iceberg-docs` need to happen against the `main` branch.
 Adjust the commands accordingly if it is not the case.
 
 #### iceberg repository preparations
 
 A PR needs to be published in `iceberg` repository with the following changes:
-1. Mark the current latest release notes to past releases
-2. Update the latest artifact links in the release notes page
-3. Add release notes for the new release version
-2. Create new folder called `docs/versioned/releases/<VERSION NUMBER>` with a `_index.md` file. See the existing folders under `docs/versioned/releases` for more details.
+
+1. Create a new folder called `docs/releases/<VERSION NUMBER>` with an `_index.md` file. See the existing folders under `docs/releases` for more details.
 
 #### iceberg-docs repository preparations
 
-A PR needs to be published in `iceberg-docs` repository with the following changes:
+A PR needs to be published in the `iceberg-docs` repository with the following changes:
 1. Update variable `latestVersions.iceberg` to the new release version in `landing-page/config.toml`
 2. Update variable `latestVersions.iceberg` to the new release version in `docs/config.toml`
+3. Mark the current latest release notes to past releases under `landing-page/content/common/release-notes.md`
+4. Update the latest artifact links in the release notes page in `landing-page/content/common/release-notes.md`
+5. Add release notes for the new release version in `landing-page/content/common/release-notes.md`
 
 #### Documentation update
```

Comment:
I know it's not part of this PR but I'm noticing that these instructions should actually come **after** the [Cutting a new version branch section](https://github.com/apache/iceberg-docs/blob/main/landing-page/content/common/how-to-release.md#cut-a-new-version-branch). These changes should then occur on that versioned branch. Here's the sequence that I think should be reflected here:

- Clone apache/iceberg
- Clone apache/iceberg-docs
- Run `cp -r iceberg/format/* iceberg-docs/landing-page/content/common/`
- cd into the `iceberg-docs` directory
- Create an `update-common-pages` branch
- Commit, Push, and Open a PR against apache/iceberg-docs (`update-common-pages` -> `main`)
- Create a branch from `main` that's named after the version (i.e. `0.13.2`). This must be done in the main repo, **not** in your fork.
- Fetch and checkout the version branch: i.e. `git fetch && git checkout 0.13.2`
- Copy the versioned docs in using `cp -r ../iceberg/docs ../iceberg-docs/docs/content/docs`
- Change into the `iceberg` directory and generate the javadoc `cd ../iceberg && ./gradlew refreshJavadoc`
- Copy the javadoc into the apache/iceberg-docs repo: `cp -r site/docs/javadoc/<VERSION NUMBER> ../iceberg-docs/javadoc`
- Change into the `iceberg-docs` directory, Commit, Push and Open a PR against the `0.13.2` branch in apache/iceberg-docs

- Checkout the main branch in apache/iceberg-docs
- Create a branch `move-forward-latest-version`
- Update variable `latestVersions.iceberg` to the new release version in `landing-page/config.toml`
- Update variable `latestVersions.iceberg` to the new release version in `docs/config.toml`
- Mark the current latest release notes to past releases under `landing-page/content/common/release-notes.md`
- Update the latest artifact links in the release notes page in `landing-page/content/common/release-notes.md`
- Add release notes for the new release version in `landing-page/content/common/release-notes.md`
- Commit, Push, and Open a PR against apache/iceberg-docs (`move-forward-latest-version` -> `main`)

Reply:
@nastra I've opened a [PR](https://github.com/nastra/iceberg-docs/pull/1) against your fork to update this to represent the flow listed above. Please take a look when you have a chance and let me know what you think.

The general ordering requires that the common pages are updated and merged and then a version branch has to be cut immediately. Everything after that (adding versioned docs & adding javadocs), then happens against that version branch.

Reply:
thanks @samredai, approved and merged it